src/smt/smt_engine.h

```diff
@@ -962,15 +962,34 @@ class CVC4_PUBLIC SmtEngine {
   bool getAbduct(const Expr& conj, Expr& abd);
 
   /**
-   * Get list of quantified formulas that were instantiated
+   * Get list of quantified formulas that were instantiated on the last call
+   * to check-sat.
    */
   void getInstantiatedQuantifiedFormulas( std::vector< Expr >& qs );
-   
+
   /**
-   * Get instantiations
+   * Get instantiations for quantified formula q.
+   *
+   * If q was a quantified formula that was instantiated on the last call to
+   * check-sat (i.e. q is returned as part of the vector in the method
+   * getInstantiatedQuantifiedFormulas above), then the list of instantiations
+   * of that formula that were generated are added to insts.
+   *
+   * In particular, if q is of the form forall x. P(x), then insts is a list
+   * of formulas of the form P(t1), ..., P(tn).
+   */
+  void getInstantiations(Expr q, std::vector<Expr>& insts);
```

Comment:
why are you doing this with expressions rather than with nodes? It introduces a level of indirection in the usage by the the single invocation module that is unnecessary. If expressions are needed for other applications of this method just have two versions?

Reply:
This is an API-level call, so it must be expressions.

Reply:
So why not have an internal-level call? This function converts nodes to expressions, which the sygus module converts back to nodes afterwards.

Reply:
There isn't any need to do an internal level call, since this is not critical to performance. Moreover, this code has the advantage of exercising the API (much like the parser does), which is good for code coverage.